test_pickle.py

```diff
@@ -7,4 +13,25 @@ def test_pickle(self):
             pickle.dump(data, f)
         with open('data.pickle', 'rb') as f:
             data = pickle.load(f)
-        assert data == {'a': 1, 'b': 2, 'c': 3}
\ No newline at end of file
+        assert data == {'a': 1, 'b': 2, 'c': 3}
+
+    def test_pickle_from_other_version(self, protocol=pickle.DEFAULT_PROTOCOL,os_type=False,version_number=False):
+
+        #If os is not given, we will use the running os and version
+        running_os_type,running_version_number = get_os_and_version()
+        if not os_type:
+            os_type = running_os_type
+        if not version_number:
+            version_number = running_version_number
+
+        # Checking how many test cases we have for the protocol
+        count = 0
+        for file in os.listdir(f"logs/{os_type}/{version_number}/protocol_{protocol}"):
```

Comment:
This part currently causes errors. We probably need some sort of check to see if the file exists or another way to make sure the test run in the right order.